### PR TITLE
Add mock auth server

### PR DESCRIPTION
**Pull request by solidgoldpig**
### What changed

- Enable running app without redis
- Add mock auth server for CI
 
Amends E2E tests
- Relax path to match auth login screen
- Only test for presence of download rather than its contents
- Prevent e2e checkAssessment helper emitting warning

Removed `DISABLE_CACHE` env var

### Why did it change

- This enables the running of e2e tests against a local instance which removes the need to wait for the review app to spin up.
- The auth server has consistently been the main cause for e2e test failures

### Issue tracking

n/a

## Screenshots

n/a

#### Automated testing

- [x] Added unit tests to cover changes
- [x] Added end-to-end tests to cover any journey changes

#### Manual testing

n/a

### Environment variables

- [x] Documented in the [README](ministryofjustice/hmpps-book-secure-move-frontend/blob/master/README.md)
- [x] Added to [continous integration](https://app.circleci.com/pipelines/github/ministryofjustice/hmpps-book-secure-move-frontend)
- [ ] Added to staging environment (deployment repository)
- [ ] Added to preproduction environment (deployment repository)
- [ ] Added to production environment (deployment repository)

### Other considerations

- [x] No new [Personally Identifiable Information (PII)](https://support.google.com/analytics/answer/6366371?hl=en) is being sent via analytics
- [x] Update [README](ministryofjustice/hmpps-book-secure-move-frontend/blob/master/README.md) with any new instructions or tasks

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -166,14 +166,25 @@ jobs:
...
+                  echo "export AUTH_PROVIDER_URL='http://localhost:3999'" >> $BASH_ENV
+                  echo "export NOMIS_ELITE2_API_URL='http://localhost:3999'" >> $BASH_ENV
+                  echo "export SERVER_HOST='localhost:3000'" >> $BASH_ENV
+                  echo "export E2E_BASE_URL_REVIEW_APP='http://localhost:3000'" >> $BASH_ENV
                   source $BASH_ENV
-                no_output_timeout: 600s
-      - run:
-          name: Run E2E tests
-          command: E2E_BASE_URL=$<<parameters.base_url_env>> npm run test-e2e:ci
+                  npm run build
```
Comment (teneightfive) [inline .circleci/config.yml:177]: Why not just run this as a separate `run` command? 

Might make it easier to read both here and in the CI steps. So something like:

```yaml
- run:
    name: Build app
    command: npm run build
- run:
    name: Start the app
    command: 
        node start.js 1>/dev/null &
        node mocks/auth-server.js &
        sleep 5
- run:
    name: Run E2E tests
    command: E2E_BASE_URL=$<<parameters.base_url_env>> npm run test-e2e:ci
```

File: .circleci/config.yml [hunk 2]
```diff
@@ -166,14 +166,25 @@ jobs:
           condition: <<parameters.use_review_app>>
           steps:
             - run:
-                name: Set Heroku Review Apps URL for Pull Requests
+                name: Run E2E tests (same instance)
                 command: |
-                  echo "export E2E_BASE_URL_REVIEW_APP=$(sh .circleci/get-review-app-url.sh)" >> $BASH_ENV
+                  echo "export REDIS_URL=''" >> $BASH_ENV
```
Comment (teneightfive) [inline .circleci/config.yml:171]: I feel like it these are no longer dynamic then we can set them on the Job using [Circle's built-in support](https://circleci.com/docs/2.0/env-vars/#setting-an-environment-variable-in-a-job) for envs.

File: .circleci/config.yml (hunk 1 above)
Comment (solidgoldpig) [reply .circleci/config.yml:177]: I tried that.

The processes get terminated. So they have to kick off in the same container process(?) as the actual e2e tests

File: .circleci/config.yml (hunk 2 above)
Comment (solidgoldpig) [reply .circleci/config.yml:171]: Or they could remain dynamic… I was just getting it working. But happy to set them as you suggest

File: .circleci/config.yml (hunk 1 above)
Comment (teneightfive) [reply .circleci/config.yml:177]: Ahhh gotcha! Makes sense

Comment (teneightfive) [reply .circleci/config.yml:177]: Could probably still split the build though?

File: .circleci/config.yml (hunk 2 above)
Comment (teneightfive) [reply .circleci/config.yml:171]: Also true. They could come from CI still. Maybe that’s better too

**Review by teneightfive — APPROVED** [review]

**Pull request merged**